### PR TITLE
[storage] TransactionStore::get_write_sets

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

In preparation for per-block SMT update. (Gonna need write sets after latest checkpoint on start up).

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?
y
## Test Plan
Unit test

### CODE REVIEW

File: storage/aptosdb/src/transaction_store/mod.rs [hunk 1]
```diff
@@ -177,6 +177,45 @@ impl TransactionStore {
...
+            begin_version < end_version,
+            "begin_version {} >= end_version {}",
+            begin_version,
+            end_version
+        );
+
+        let mut iter = self.db.iter::<WriteSetSchema>(Default::default())?;
+        iter.seek(&begin_version)?;
+
+        let mut ret = Vec::with_capacity(end_version.saturating_sub(begin_version) as usize);
```
Comment (lightmark) [inline storage/aptosdb/src/transaction_store/mod.rs:201]: too verbose...

**Review by lightmark — APPROVED** [review]

File: storage/aptosdb/src/transaction_store/mod.rs [hunk 2]
```diff
@@ -177,6 +177,45 @@ impl TransactionStore {
...
     }
 
+    /// Get write sets in `[begin_version, end_version)` half-open range.
+    ///
+    /// N.b. an empty `Vec` is returned when `begin_version == end_version`
+    pub fn get_write_sets(
+        &self,
+        begin_version: Version,
+        end_version: Version,
+    ) -> Result<Vec<WriteSet>> {
```
Comment (sitalkedia) [inline storage/aptosdb/src/transaction_store/mod.rs:187]: I am wondering if it's better to return an iterator for this API, just like the `get_transaction_iter` API? That way, we can avoid allocating a large amount of memory if this `WriteSet` range is too big?

Comment (msmouse) [reply storage/aptosdb/src/transaction_store/mod.rs:187]: I think at this point it's an over optimization. 😆

**Pull request closed without merging**